colin-api/src/colin_api/models/business.py

```diff
@@ -671,3 +688,66 @@ def reset_corp_states(cls, cursor, event_ids: List):
         except Exception as err:
             current_app.logger.error(f'Error in Business: Failed reset ended corp_state rows for events {event_ids}')
             raise err
+
+    @classmethod
+    def find_by_corp_num(cls, corp_num: str, con=None) -> Business:
+        """Return a Business by corp_num."""
+        business = None
+        try:
+            # get record
+            if not con:
+                con = DB.connection
+                # con.begin()
+
+            cursor = con.cursor()
+            cursor.execute(
+                """
+                select corp.corp_num, corp_typ_cd, can_jur_typ_cd, othr_juris_desc, home_juris_num
+                from CORPORATION corp
+                    left join JURISDICTION on JURISDICTION.corp_num = corp.corp_num
+                where corp.CORP_NUM=:corp_num
+                """,
+                corp_num=corp_num
+            )
+            business = cursor.fetchone()
+            if not business:
+                raise BusinessNotFoundException(corp_num=corp_num)
```

Comment:
Update this line to `raise BusinessNotFoundException(identifier=corp_num)`.  

`BusinessNotFoundException` has specific logic inside to look for the `identifier` param passed in.  This is key to the GET business resource code for the new endpoint to catch a `GenericException` so it can return a 404 to the api consumer.

Reply:
I updated